packages/core/src/render3/instructions/debug.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+
+/**
+ * Creates an `Array` construction with a given name. This is useful when 
+ * looking for memory consumption to see what time of array it is.
+ * 
+ * 
+ * @param name Name to give to the constructor
+ * @returns A subclass of `Array` if possible. This can only be done in 
+ *          environments which support `class` construct.
+ */
+export function makeNamedArray(name: string): typeof Array {
+  try {
+    // We need to do it this way so that TypeScript does not down-level the below code.
+    return (new Function('Array', `return class ${name} extends Array{}`))(Array);
```

Comment:
You're going to want to add a `Symbol.species` here, if you don't want to effect the semantics

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/species]

```suggestion
    return (new Function('Array', `return class ${name} extends Array{ get [Symbol.species]() { return Array; } }`))(Array);
```

Reply:
Cool I did not know such a thing existed, learned something new. 

I read the link and I think I specifically don't want this, since we do want to propagate the array type. Did you have something else in mind.

Reply:
:+1: